lib/airbrake/rack/user.rb

```diff
@@ -17,9 +17,15 @@ def self.extract(rack_env)
         end
 
         # Fallback mode (OmniAuth support included). Works only for Rails.
+        user = try_current_user(rack_env)
+        new(user) if user
+      end
+
+      def self.try_current_user(rack_env)
         controller = rack_env['action_controller.instance']
         return unless controller.respond_to?(:current_user)
-        new(controller.current_user) if controller.current_user
+        return unless [-1, 0].include?(controller.method(:current_user).arity)
```

Comment:
I think the condition I suggested is wrong, but I hope you got the idea (no need to use #include and array since we can rewrite it, so it uses comparison).

 ```rb
return if controller.method(:current_user).arity != 0
```